src/pages/ProcessDetail.tsx

```diff
@@ -480,91 +399,70 @@ class ProcessDetail extends React.PureComponent<IProps, IState> {
-            <FormattedMessage id={`process.tabs.${tab}`} />
+    const renderTab = (tabParam: string, selectedTabParam: string) => (
+        <span
+            id={tabParam}
+            key={tabParam}
+            className={tabParam === selectedTabParam ? "active" : ""}
+            onClick={switchTab(tabParam)}
+        >
+            <FormattedMessage id={`process.tabs.${tabParam}`} />
         </span>
     );
 
-    addConfirmDialogActions = ({ showConfirmDialog }: ConfirmDialogActions) => {
-        if (this.state.showConfirmDialog !== showConfirmDialog) {
-            this.setState({ showConfirmDialog });
-        }
-        return <></>;
-    };
-
-    render() {
-        const {
-            loaded,
-            notFound,
-            process,
-            tabs,
-            stepUserInput,
-            selectedTab,
-            subscriptionProcesses,
-            isActionsMenuOnScreen,
-        } = this.state;
-        if (!process) {
-            return null;
-        }
-
-        const step = process.steps.find((step: Step) => step.status === "pending");
-        const renderNotFound = loaded && notFound;
-        const renderContent = loaded && !notFound;
-        return (
-            <EuiPage css={processDetailStyling}>
-                <RunningProcessesContext.Consumer>
-                    {(rpc: any) => this.handleUpdateProcess(rpc.runningProcesses)}
-                </RunningProcessesContext.Consumer>
-                <ConfirmationDialogContext.Consumer>
-                    {(cdc) => this.addConfirmDialogActions(cdc)}
-                </ConfirmationDialogContext.Consumer>
-                <EuiPageBody component="div" className="mod-process-detail">
-                    <section className="tabs">{tabs.map((tab) => this.renderTab(tab, selectedTab))}</section>
-                    {renderContent &&
-                        this.renderTabContent(selectedTab, process, step, stepUserInput, subscriptionProcesses)}
-                    {renderContent &&
-                        !isActionsMenuOnScreen &&
-                        this.renderFixedTabMenu(selectedTab, process, step, stepUserInput)}
-                    {renderNotFound && (
-                        <section className="not-found">
-                            <EuiPanel>
-                                <h1>
-                                    <FormattedMessage id="process.notFound" />
-                                </h1>
-                            </EuiPanel>
-                        </section>
-                    )}
-                    <ScrollUpButton />
-                </EuiPageBody>
-            </EuiPage>
-        );
+    if (!process) {
+        return null;
     }
-}
 
-ProcessDetail.contextType = ApplicationContext;
+    const step = process.steps.find((step: Step) => step.status === "pending");
+    const renderNotFound = loaded && notFound;
+    const renderContent = loaded && !notFound;
+    handleUpdateProcess(runningProcesses);
```

Comment:
I believe this needs to be in a `useffect` to handle recall the function when `runningProcesses` updates.

Reply:
works without useeffect.

Reply:
So, should I keep it like that ?